editor/enable-tracking-prompt/index.js

```diff
@@ -18,48 +20,89 @@ import { removeNotice } from '../actions';
+	constructor() {
+		super();
+		this.state = {
+			showInfoPopover: false,
+		};
+		this.dismissTrackingPrompt = this.dismissTrackingPrompt.bind( this );
+		this.toggleInfoPopover = this.toggleInfoPopover.bind( this );
+		this.handleClickOutside = this.handleClickOutside.bind( this );
+	}
+
+	dismissTrackingPrompt( enableTracking ) {
 		window.setUserSetting(
 			'gutenberg_tracking',
 			enableTracking ? 'on' : 'off'
 		);
 		if ( enableTracking ) {
 			bumpStat( 'tracking', 'opt-in' );
 		}
-		props.removeNotice( TRACKING_PROMPT_NOTICE_ID );
+		this.props.removeNotice( TRACKING_PROMPT_NOTICE_ID );
 	}
 
-	return (
-		<div className="enable-tracking-prompt">
-			<div className="enable-tracking-prompt__buttons">
-				<Button
-					isPrimary
-					isSmall
-					onClick={ () => dismissTrackingPrompt( true ) }
-				>
-					{ __( 'Yes' ) }
-				</Button>
-				<Button
-					isSecondary
-					isSmall
-					onClick={ () => dismissTrackingPrompt( false ) }
-				>
-					{ __( 'No' ) }
-				</Button>
-			</div>
-			<div className="enable-tracking-prompt__message">
-				{ __( 'Can Gutenberg collect data about your usage of the editor?' ) }
-			</div>
-			<div className="enable-tracking-prompt__clarification">
-				{ __( 'Usage data is completely anonymous, does not include your post content, and will only be used to improve the editor.' ) }
+	toggleInfoPopover( event ) {
+		event.stopPropagation();
+		this.setState( {
+			showInfoPopover: ! this.state.showInfoPopover,
+		} );
+	}
+
+	handleClickOutside() {
+		this.setState( {
+			showInfoPopover: false,
+		} );
+	}
+
+	render() {
+		return (
+			// Ignore reason: This event handler exists only to catch click
+			// events in the notice but outside the "More info" popover.
+			// eslint-disable-next-line jsx-a11y/click-events-have-key-events, jsx-a11y/no-static-element-interactions, jsx-a11y/onclick-has-role
+			<div
+				className="enable-tracking-prompt"
+				onClick={ this.handleClickOutside }
+			>
+				<p className="enable-tracking-prompt__message">
+					{ __( 'Can Gutenberg collect data about your usage of the editor?' ) }
+				</p>
+				<p className="enable-tracking-prompt__buttons">
+					<Button
+						isPrimary
+						isSmall
+						onClick={ () => this.dismissTrackingPrompt( true ) }
+					>
+						{ __( 'Yes' ) }
+					</Button>
+					<Button
+						isSecondary
+						isSmall
+						onClick={ () => this.dismissTrackingPrompt( false ) }
+					>
+						{ __( 'No' ) }
+					</Button>
+					<Button
+						className="enable-tracking-prompt__more-info"
+						onClick={ this.toggleInfoPopover }
+					>
+						<span>{ __( 'More info' ) }</span>
+						{ this.state.showInfoPopover && (
+							<Popover
+								position="bottom right"
+								className="enable-tracking-prompt__clarification"
+							>
+								{ __( 'Usage data is completely anonymous, does not include your post content, and will only be used to improve the editor.' ) }
+							</Popover>
+						) }
+					</Button>
```

Comment:
I'd move the popover out from the button and use `aria-expanded` on the button based on the showInfoPopover state. See for example post-visibility.